# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/shell/js.cpp
# Commit: 81ac1990bb44
# Full Hash: 81ac1990bb44e81a2146c84d6b226408e0df0891
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2023-06-15 21:43:34
# Description:
#   Bug 1838166 - Don't allow changing code write protection at runtime. r=nbp
#   
#   To support this we'd have to reprotect all of the allocated memory pages.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180923
# ==============================================================================

diff -r e31965aaf77a -r 81ac1990bb44 js/src/shell/js.cpp
--- a/js/src/shell/js.cpp	Thu Jun 15 16:31:28 2023 +0000
+++ b/js/src/shell/js.cpp	Thu Jun 15 16:31:36 2023 +0000
@@ -4699,6 +4699,19 @@
     }
   }
 
+  // Changing code memory protection settings at runtime is not supported. Don't
+  // throw if not changing the setting because some jit-tests depend on that.
+  if (opt == JSJITCOMPILER_WRITE_PROTECT_CODE) {
+    uint32_t writeProtect;
+    MOZ_ALWAYS_TRUE(JS_GetGlobalJitCompilerOption(
+        cx, JSJITCOMPILER_WRITE_PROTECT_CODE, &writeProtect));
+    if (bool(number) != writeProtect) {
+      JS_ReportErrorASCII(cx, "Can't change code write protection at runtime");
+      return false;
+    }
+    return true;
+  }
+
   // Throw if trying to disable all the Wasm compilers.  The logic here is that
   // if we're trying to disable a compiler that is currently enabled and that is
   // the last compiler enabled then we must throw.
